mount/mount_linux.go

```diff
@@ -292,18 +298,66 @@ func (mounter *SafeFormatAndMount) checkAndRepairXfsFilesystem(source string) er
 			return nil
 		} else {
 			klog.Warningf("Filesystem corruption was detected for %s, running xfs_repair to repair", source)
-			out, err := mounter.Exec.Command("xfs_repair", args...).CombinedOutput()
-			if err != nil {
-				return NewMountError(HasFilesystemErrors, "'xfs_repair' found errors on device %s but could not correct them: %s\n", source, out)
-			} else {
-				klog.Infof("Device %s has errors which were corrected by xfs_repair.", source)
-				return nil
+			var uncorrectedErr error
+
+			// If xfs_repair fails to repair the file system successfully, try giving the same xfs_repair
+			// command twice more; xfs_repair may be able to make more repairs on successive runs.
+			for i := 0; i < xfsRepairRetries; i++ {
```

Comment:
Dumb question but I don't understand this much so bear with me: this code seems to retry 3 times no mater what, is it possible that after the first or second try, subsequent retries are unnecessary?

Reply:
Won't this result in calling `xfs_repair` 4 times? We already called one time outside the loop.

Reply:
the first attempt is only a check (with -n flag) to see if there is any fs issues, it will not do any repair, and it is faster. So it can save some time if the fs is error free.